enginetest/queries/script_queries.go

```diff
@@ -2273,6 +2273,55 @@ var ScriptTests = []ScriptTest{
 			},
 		},
 	},
+	{
+		Name: "having clause without groupby clause, all rows implicitly form a single aggregate group",
+		SetUpScript: []string{
+			"create table numbers (val int);",
+			"insert into numbers values (1), (2), (3);",
+			"insert into numbers values (2), (4);",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query:    "select val from numbers;",
+				Expected: []sql.Row{{1}, {2}, {3}, {2}, {4}},
+			},
+			{
+				Query:    "select val as a from numbers having a = val;",
+				Expected: []sql.Row{{1}, {2}, {3}, {2}, {4}},
+			},
+			{
+				Query:    "select val as a from numbers group by val having a = val;",
+				Expected: []sql.Row{{1}, {2}, {3}, {4}},
+			},
+			{
+				Query:    "select val as a from numbers as t1 group by t1.val having a = t1.val;",
+				Expected: []sql.Row{{1}, {2}, {3}, {4}},
+			},
+			{
+				Query:    "select t1.val as a from numbers as t1 group by 1 having a = t1.val;",
+				Expected: []sql.Row{{1}, {2}, {3}, {4}},
+			},
+			{
+				Query:    "select t1.val as a from numbers as t1 having a = t1.val;",
+				Expected: []sql.Row{{1}, {2}, {3}, {2}, {4}},
+			},
+			{
+				Query:    "select count(*) from numbers having count(*) = 5;",
+				Expected: []sql.Row{{5}},
+			},
+			{
+				// MySQL returns `Unknown column 'val' in 'having clause'` error for this query,
+				// but GMS builds GroupBy for any aggregate function.
+				Skip:  true,
+				Query: "select count(*) from numbers having count(*) > val;",
```

Comment:
Good find! This is an interesting test case. If an alias is applied to the first `count(*)`, then MySQL is happy and will execute the query. Seems like it might really be the aggregate function in the having clause that is throwing MySQL off? Good skipped test to have in the suite.